### PR TITLE
Add code to finish collecting outputs and to plot pub figures

### PR DESCRIPTION
**Pull request by taylorreiter**
Adds the final script to combine all of the outputs into a single table (with lots of checks to make sure there aren't duplicates sneaking in anywhere), and plotting code. I know there are commented paths in one of the scripts -- it hasn't finished running on snakemake yet and having those there makes it easier to debug if a problem comes up.

### CODE REVIEW

File: Snakefile [hunk 1]
```diff
@@ -2,7 +2,24 @@ configfile: "snakemake_config_blast.yml"
 
 rule all:
     input:
-        "outputs/all_outputs_combined.tsv"
+        fig2b = "figures/figure2b.pdf",
```
Comment (mertcelebi) [inline Snakefile:5]: Skipping `fib1a` is intentional to match the pub I assume, right?

File: Snakefile [hunk 2]
```diff
@@ -309,12 +326,39 @@ rule combine_foldseek_outputs:
 
 rule combine_all_outputs:
     input:
+        blast = "outputs/blast/blastp_results_empty_rm.out",
```
Comment (mertcelebi) [inline Snakefile:329]: where does this come from? I assume it's the output of the `blast.snakefile` + you removing empty files, but could you document it?

File: Snakefile [hunk 3]
```diff
@@ -309,12 +326,39 @@ rule combine_foldseek_outputs:
...
         avg_pid = "outputs/mean_pid/all_avg_pid.tsv",
         all_features = "outputs/shared_feature_residues/3_shared_residue_summaries_combined/all_shared_residues_combined.tsv",
         all_hmm = "outputs/hmm/hmmscan/all-hmmscan-tbl-out.tsv",
         all_fsk  = "outputs/foldseek/foldseek/all_foldseek.tsv",
         class_map = "inputs/uniprot_gene_name_to_class_map.tsv"
-    output: all_outputs = "outputs/all_outputs_combined.tsv"
+    output: all_outputs = "outputs/all_outputs_summarized.tsv"
     conda: "envs/tidyverse.yml"
     benchmark: "benchmarks/combine_all_outputs.txt"
     script: "snakemake/snakemake_combine_all_outputs.R"
```
Comment (mertcelebi) [inline Snakefile:338]: nit:
```suggestion
    script: "snakemake/snakemake_summarize_all_outputs.R"
```

File: snakemake/snakemake_combine_all_outputs.R [hunk 4]
```diff
@@ -1,4 +1,67 @@
...
 library(readr)
 
-# placeholder for script that I'll generate after the rest of the code is done running 
+# read in blast results that seeded the snakemake run.
+# this table has the accessions that failed download with entrez error code shown below removed:
+# Invalid uid * at position= 0
+# this table will be used to filter the rest of the results to make sure nothing sneaks in,
+# and to count the total number of results that ended up being generated by each process
+blast_results <- read_tsv(snakemake@input[['blast']])
+#blast_results <- read_tsv("outputs/blast/blastp_results_empty_rm.out")
```
Comment (mertcelebi) [inline snakemake/snakemake_combine_all_outputs.R:10]: boop

File: snakemake/snakemake_combine_all_outputs.R [hunk 5]
```diff
@@ -1,4 +1,67 @@
...
+
+# this is a hand-curated file that uses the uniprot protein names to inform what class each actin sequence falls into
+class_map <- read_csv(snakemake@input[['class_map']]) %>%
+#class_map <- read_tsv("inputs/uniprot_gene_name_to_class_map.tsv") %>%
```
Comment (mertcelebi) [inline snakemake/snakemake_combine_all_outputs.R:14]: bap

File: snakemake/snakemake_combine_all_outputs.R [hunk 6]
```diff
@@ -1,4 +1,67 @@
 library(dplyr)
 library(readr)
 
-# placeholder for script that I'll generate after the rest of the code is done running 
+# read in blast results that seeded the snakemake run.
+# this table has the accessions that failed download with entrez error code shown below removed:
```
Comment (mertcelebi) [inline snakemake/snakemake_combine_all_outputs.R:5]: is "entrez error code" a thing? what does that mean? first i am hearing

File: snakemake/snakemake_combine_all_outputs.R [hunk 7]
```diff
@@ -1,4 +1,67 @@
...
+#blast_results <- read_tsv("outputs/blast/blastp_results_empty_rm.out")
+
+# this is a hand-curated file that uses the uniprot protein names to inform what class each actin sequence falls into
+class_map <- read_csv(snakemake@input[['class_map']]) %>%
+#class_map <- read_tsv("inputs/uniprot_gene_name_to_class_map.tsv") %>%
+  filter(!is.na(class))
+
+# read in average percent identity results
+avg_pid <- read_tsv(snakemake@input[['avg_pid']]) %>%
+#avg_pid <- read_tsv("outputs/mean_pid/all_avg_pid.tsv") %>%
```
Comment (mertcelebi) [inline snakemake/snakemake_combine_all_outputs.R:19]: hello

File: snakemake/snakemake_combine_all_outputs.R [hunk 8]
```diff
@@ -1,4 +1,67 @@
...
+         prot2 = gsub("\\.1\\|.*",  "\\.1", prot2),
+         prot2 = gsub("\\.2\\|.*",  "\\.2", prot2),
+         prot2 = gsub("\\.3\\|.*",  "\\.3", prot2),
+         prot2 = gsub("\\|", "_", prot2)) %>%
+  filter(prot2 %in% blast_results$sseqid) %>%
+  filter(!prot2 %in% c("NP_187818.1", "BAN39743.1", "TMW59056.1")) # filter three results where we detected inaccuracies
+  
+# read in the shared feature residue results (lateral contact, longitudinal contats, and ATP-binding residues)
+features <- read_tsv(snakemake@input[['all_features']]) %>%
+#features <- read_tsv("outputs/shared_feature_residues/3_shared_residue_summaries_combined/all_shared_residues_combined.tsv") %>%
```
Comment (mertcelebi) [inline snakemake/snakemake_combine_all_outputs.R:36]: hi

File: snakemake/snakemake_combine_all_outputs.R [hunk 9]
```diff
@@ -1,4 +1,67 @@
...
+  distinct() %>%
+  filter(protein %in% blast_results$sseqid)
+
+# read in the hmm results
+hmm <- read_tsv(snakemake@input[['all_hmm']]) %>%
+#hmm <- read_tsv("outputs/hmm/hmmscan/all-hmmscan-tbl-out.tsv") %>%
```
Comment (mertcelebi) [inline snakemake/snakemake_combine_all_outputs.R:42]: one more

**Review by mertcelebi — APPROVED** [review]
Looks good, a couple small nits. Figure code is a little tough to review, so I guess it'll be easier to tell once they're fully generated. Will you upload the figures to the repo as well?

File: Snakefile (hunk 2 above)
Comment (taylorreiter) [reply Snakefile:329]: ya i'll add this script in!

File: Snakefile (hunk 1 above)
Comment (taylorreiter) [reply Snakefile:5]: ya -- the figures that are skipped are ones that are diagrams in the pub, not generated by code

File: Snakefile (hunk 3 above)
Comment (taylorreiter) [reply Snakefile:338]: good catch, i'll change this

File: snakemake/snakemake_combine_all_outputs.R (hunk 6 above)
Comment (taylorreiter) [reply snakemake/snakemake_combine_all_outputs.R:5]: this is what caused the empty files. i've documented it in 3 places in the repo...I tried the API in 3 different ways and it consistently failed, so i think just documenting that fact is good enough for now.

**Pull request merged**